var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -387,6 +387,11 @@ def cmake_args(self):
                 '-DMPI_CXX_COMPILER=%s' % spec['mpi'].mpicxx,
                 '-DMPI_Fortran_COMPILER=%s' % spec['mpi'].mpifc,
             ])
+            if '+cuda' in spec:
+                options.extend([
+                    '-DDEAL_II_MPI_WITH_CUDA_SUPPORT:BOOL=ON',
+                    '-DCUDA_HOST_COMPILER=%s' % spec['mpi'].mpicxx
+                ])
```

Comment:
I wasn't aware of the [`DEAL_II_MPI_WITH_CUDA_SUPPORT`](https://github.com/dealii/dealii/blob/159efe444170a59298b43472bbf7184028b735ab/cmake/configure/configure_1_mpi.cmake#L60) flag. Glad that you found it!

In #19253, we're updating the package to use a newer syntax that the `CMakePackage` offers. Perhaps you'd be willing to follow suite? So I think that this block would then read
```
options.extend([
    self.define('DEAL_II_MPI_WITH_CUDA_SUPPORT, True),
    self.define('CUDA_HOST_COMPILER', spec['mpi'].mpicxx)
])
```

Reply:
I don't know if you want to set that flag (`DEAL_II_MPI_WITH_CUDA_SUPPORT`). When that flag is ON we assume that MPI was compiled with CUDA support. This avoids copy to the host but if MPI was compiled without CUDA support the code will crash.

Reply:
This project moves really fast - making git rebase indispensable.  This update uses the new syntax, and I've tested it's working on my system.

I should note that I monkey-patched Sacado locally to get this to compile (https://github.com/trilinos/Trilinos/issues/8174), but that's not deal.II's fault.

Reply:
> This project moves really fast - making git rebase indispensable. This update uses the new syntax, and I've tested it's working on my system.

You were just a bit unlucky that I happened to pick this weekend to make a few touch-ups, and then that snowballed into a complete revamp of the package :-D Thanks for updating the changes that you made, and for retesting them!